include/util/coordinate_calculation.hpp

```diff
@@ -25,16 +25,49 @@ const constexpr long double EARTH_RADIUS = 6372797.560856;
 
 
 //! Takes the squared euclidean distance of the input coordinates. Does not return meters!
-std::uint64_t squaredEuclideanDistance(const Coordinate &lhs, const Coordinate &rhs);
+std::uint64_t squaredEuclideanDistance(const Coordinate lhs, const Coordinate rhs);
 
 double haversineDistance(const Coordinate first_coordinate, const Coordinate second_coordinate);
 
 double greatCircleDistance(const Coordinate first_coordinate, const Coordinate second_coordinate);
 
-std::pair<double, FloatCoordinate>
-projectPointOnSegment(const FloatCoordinate &projected_xy_source,
-                      const FloatCoordinate &projected_xy_target,
-                      const FloatCoordinate &projected_xy_coordinate);
+inline std::pair<double, FloatCoordinate> projectPointOnSegment(const FloatCoordinate &source,
+                                                         const FloatCoordinate &target,
+                                                         const FloatCoordinate &coordinate)
```

Comment:
Should we pass all coordinate types by value? They're wrappers around 2*64bit (`FloatCoordinate`) and 2*32bit (`Coordinate`). Looking at the coordinates why is `FloatLongitude` and `FloatLatitude` `double` anyway, do we really need the precision?